app/controllers/admin/product_import_controller.rb

```diff
@@ -0,0 +1,52 @@
+require 'roo'
+
+class Admin::ProductImportController < Spree::Admin::BaseController
+
+  before_filter :check_upload, except: :index
+
+  def import
+    # Save uploaded file to tmp directory
+    @filepath = save_upload(params[:file])
+    @importer = ProductImporter.new(File.new(@filepath), editable_enterprises)
+
+    if @importer.errors.present?
+      flash[:notice] = @importer.errors.full_messages.to_sentence
+    end
+  end
+
+  def save
+    file = File.new(params[:filepath])
```

Comment:
Can always create a cleanup job using `whenever` that runs each night/week/month...